src/Miso/Style.hs

```diff
@@ -0,0 +1,1357 @@
+  , insetInlineEnd
+  , insetInlineStart
+  , justifyContent
+  , justifyItems
+  , justifySelf
+  , left
+  , letterSpacing
+  , linearCrossGravity
+  , linearDirection
+  , linearGravity
+  , linearLayoutGravity
+  , linearWeight
+  , linearWeightSum
+  , lineHeight
+  , marginBottom
+  , marginInlineEnd
+  , marginInlineStart
+  , marginLeft
+  , margin
+  , marginRight
+  , marginTop
+  , maskImage
+  , mask
+  , maxHeight
+  , maxWidth
+  , minHeight
+  , minWidth
+  , opacity
+  , order
+  , overflow
+  , overflowX
+  , overflowY
+  , paddingBottom
+  , paddingInlineEnd
+  , paddingInlineStart
+  , paddingLeft
+  , padding
+  , paddingRight
+  , paddingTop
+  , perspective
+  , position
+  , relativeAlignBottom
+  , relativeAlignInlineEnd
+  , relativeAlignInlineStart
+  , relativeAlignLeft
+  , relativeAlignRight
+  , relativeAlignTop
+  , relativeBottomOf
+  , relativeCenter
+  , relativeId
+  , relativeInlineEndOf
+  , relativeInlineStartOf
+  , relativeLayoutOnce
+  , relativeLeftOf
+  , relativeRightOf
+  , relativeTopOf
+  , right
+  , rowGap
+  , textAlign
+  , textDecoration
+  , textIndent
+  , textOverflow
+  , textShadow
+  , textStrokeColor
+  , textStroke
+  , textStrokeWidth
+  , top
+  , transform
+  , transformOrigin
+  , transitionDelay
+  , transitionDuration
+  , transition
+  , transitionProperty
+  , transitionTimingFunction
+  , verticalAlign
+  , visibility
+  , whiteSpace
+  , width
+  , wordBreak
+  , xAutoFontSize
+  , xAutoFontSizePresetSizes
+  , xHandleColor
+  , xHandleSize
+  , zIndex
+  -- *** Colors
+  , module Miso.Style.Color
+  ) where
+-----------------------------------------------------------------------------
+import           Data.Map (Map)
+import qualified Data.Map as M
+import           Miso.String (MisoString)
+import qualified Miso.String as MS
+import           Miso.Style.Color
+import           Miso.Property
+import           Miso.Types (Attribute)
+import qualified Miso.Types as MT
+-----------------------------------------------------------------------------
+import           Prelude hiding (filter)
+-----------------------------------------------------------------------------
+-- | Smart constructor for Attributes. This function is helpful when 
+-- constructing 'Style'.
+--
+-- Example shown below.
+--
+-- @
+-- div_ [ style_  [ "background" =: "red" ] []
+-- @
+--
+(=:) :: k -> v -> (k, v)
+k =: v = (k,v)
+-----------------------------------------------------------------------------
+-----------------------------------------------------------------------------
+-- | 'Style'
+--
+-- Type for a CSS 'Style'
+--
+type Style = (MisoString, MisoString)
+-----------------------------------------------------------------------------
+-- | 'Styles'
+--
+-- Type for a @Map@ of CSS 'Style'. Used with @StyleSheet@.
+--
+type Styles = Map MisoString MisoString
+-----------------------------------------------------------------------------
+-- | 'StyleSheet'
+--
+-- Type for a CSS style on native
+--
```

Comment:
It's Map from MisoString to Styles, which itself is Map from MisoString to Miso String.
Maybe add a sentence about what the keys / values at each level are? The same for sheet_ below.